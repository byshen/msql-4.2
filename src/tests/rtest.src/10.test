# Limit testing

create table limit_test (
	int_val 	int, 
	char_val 	char(10), 
	real_val 	real,
	date_val	date,
	text_val	text(5)
) \p\g

insert into limit_test values (1, '1', 1.1, '1-Jan-2000', 'zzzzzz1') \p\g
insert into limit_test values (2, '2', 1.2, '2-Jan-2000', 'zzzzzz2') \p\g
insert into limit_test values (3, '3', 1.3, '3-Jan-2000', 'zzzzzz3') \p\g
insert into limit_test values (4, '4', 1.4, '4-Jan-2000', 'zzzzzz4') \p\g
insert into limit_test values (5, '5', 1.5, '5-Jan-2000', 'zzzzzz5') \p\g
insert into limit_test values (6, '6', 1.6, '6-Jan-2000', 'zzzzzz6') \p\g
insert into limit_test values (7, '7', 1.7, '7-Jan-2000', 'zzzzzz7') \p\g
insert into limit_test values (8, '8', 1.8, '8-Jan-2000', 'zzzzzz8') \p\g
insert into limit_test values (9, '9', 1.9, '9-Jan-2000', 'zzzzzz9') \p\g

insert into limit_test values (10, '10', 1.10, '10-Jan-2000', 'zzzzzz10') \p\g
insert into limit_test values (11, '11', 1.11, '11-Jan-2000', 'zzzzzz11') \p\g
insert into limit_test values (12, '12', 1.12, '12-Jan-2000', 'zzzzzz12') \p\g
insert into limit_test values (13, '13', 1.13, '13-Jan-2000', 'zzzzzz13') \p\g
insert into limit_test values (14, '14', 1.14, '14-Jan-2000', 'zzzzzz14') \p\g
insert into limit_test values (15, '15', 1.15, '15-Jan-2000', 'zzzzzz15') \p\g
insert into limit_test values (16, '16', 1.16, '16-Jan-2000', 'zzzzzz16') \p\g
insert into limit_test values (17, '17', 1.17, '17-Jan-2000', 'zzzzzz17') \p\g
insert into limit_test values (18, '18', 1.18, '18-Jan-2000', 'zzzzzz18') \p\g
insert into limit_test values (19, '19', 1.19, '19-Jan-2000', 'zzzzzz19') \p\g

insert into limit_test values (20, '20', 1.20, '20-Jan-2000', 'zzzzzz20') \p\g
insert into limit_test values (21, '21', 1.21, '21-Jan-2000', 'zzzzzz21') \p\g
insert into limit_test values (22, '22', 1.22, '22-Jan-2000', 'zzzzzz22') \p\g
insert into limit_test values (23, '23', 1.23, '23-Jan-2000', 'zzzzzz23') \p\g
insert into limit_test values (24, '24', 1.24, '24-Jan-2000', 'zzzzzz24') \p\g
insert into limit_test values (25, '25', 1.25, '25-Jan-2000', 'zzzzzz25') \p\g
insert into limit_test values (26, '26', 1.26, '26-Jan-2000', 'zzzzzz26') \p\g
insert into limit_test values (27, '27', 1.27, '27-Jan-2000', 'zzzzzz27') \p\g
insert into limit_test values (28, '28', 1.28, '28-Jan-2000', 'zzzzzz28') \p\g
insert into limit_test values (29, '29', 1.29, '29-Jan-2000', 'zzzzzz29') \p\g

# Do some selects

select * from limit_test limit 5\p\g
select * from limit_test order by real_val desc limit 7\p\g
select text_val from limit_test limit 10\p\g
select text_val from limit_test limit 100\p\g
