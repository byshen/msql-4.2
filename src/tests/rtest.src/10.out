
Welcome to the miniSQL monitor.  Type \h for help.


mSQL >     ->     ->     ->     ->     ->     ->     -> 
Query buffer
------------

create table limit_test (
	int_val 	int, 
	char_val 	char(10), 
	real_val 	real,
	date_val	date,
	text_val	text(5)
) 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL >     -> 
Query buffer
------------
insert into limit_test values (1, '1', 1.1, '1-Jan-2000', 'zzzzzz1') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into limit_test values (2, '2', 1.2, '2-Jan-2000', 'zzzzzz2') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into limit_test values (3, '3', 1.3, '3-Jan-2000', 'zzzzzz3') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into limit_test values (4, '4', 1.4, '4-Jan-2000', 'zzzzzz4') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into limit_test values (5, '5', 1.5, '5-Jan-2000', 'zzzzzz5') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into limit_test values (6, '6', 1.6, '6-Jan-2000', 'zzzzzz6') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into limit_test values (7, '7', 1.7, '7-Jan-2000', 'zzzzzz7') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into limit_test values (8, '8', 1.8, '8-Jan-2000', 'zzzzzz8') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into limit_test values (9, '9', 1.9, '9-Jan-2000', 'zzzzzz9') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL >     -> 
Query buffer
------------
insert into limit_test values (10, '10', 1.10, '10-Jan-2000', 'zzzzzz10') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into limit_test values (11, '11', 1.11, '11-Jan-2000', 'zzzzzz11') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into limit_test values (12, '12', 1.12, '12-Jan-2000', 'zzzzzz12') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into limit_test values (13, '13', 1.13, '13-Jan-2000', 'zzzzzz13') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into limit_test values (14, '14', 1.14, '14-Jan-2000', 'zzzzzz14') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into limit_test values (15, '15', 1.15, '15-Jan-2000', 'zzzzzz15') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into limit_test values (16, '16', 1.16, '16-Jan-2000', 'zzzzzz16') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into limit_test values (17, '17', 1.17, '17-Jan-2000', 'zzzzzz17') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into limit_test values (18, '18', 1.18, '18-Jan-2000', 'zzzzzz18') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into limit_test values (19, '19', 1.19, '19-Jan-2000', 'zzzzzz19') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL >     -> 
Query buffer
------------
insert into limit_test values (20, '20', 1.20, '20-Jan-2000', 'zzzzzz20') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into limit_test values (21, '21', 1.21, '21-Jan-2000', 'zzzzzz21') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into limit_test values (22, '22', 1.22, '22-Jan-2000', 'zzzzzz22') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into limit_test values (23, '23', 1.23, '23-Jan-2000', 'zzzzzz23') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into limit_test values (24, '24', 1.24, '24-Jan-2000', 'zzzzzz24') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into limit_test values (25, '25', 1.25, '25-Jan-2000', 'zzzzzz25') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into limit_test values (26, '26', 1.26, '26-Jan-2000', 'zzzzzz26') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into limit_test values (27, '27', 1.27, '27-Jan-2000', 'zzzzzz27') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into limit_test values (28, '28', 1.28, '28-Jan-2000', 'zzzzzz28') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into limit_test values (29, '29', 1.29, '29-Jan-2000', 'zzzzzz29') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL >     ->     -> 
Query buffer
------------

select * from limit_test limit 5
[continue]
    -> 
Query OK.  5 row(s) modified or retrieved.

 +----------+------------+--------------+-------------+----------+
 | int_val  | char_val   | real_val     | date_val    | text_val |
 +----------+------------+--------------+-------------+----------+
 | 1        | 1          | 1.1          | 01-Jan-2000 | zzzzzz1  |
 | 2        | 2          | 1.2          | 02-Jan-2000 | zzzzzz2  |
 | 3        | 3          | 1.3          | 03-Jan-2000 | zzzzzz3  |
 | 4        | 4          | 1.4          | 04-Jan-2000 | zzzzzz4  |
 | 5        | 5          | 1.5          | 05-Jan-2000 | zzzzzz5  |
 +----------+------------+--------------+-------------+----------+



mSQL > 
Query buffer
------------
select * from limit_test order by real_val desc limit 7
[continue]
    -> 
Query OK.  7 row(s) modified or retrieved.

 +----------+------------+--------------+-------------+----------+
 | int_val  | char_val   | real_val     | date_val    | text_val |
 +----------+------------+--------------+-------------+----------+
 | 9        | 9          | 1.9          | 09-Jan-2000 | zzzzzz9  |
 | 8        | 8          | 1.8          | 08-Jan-2000 | zzzzzz8  |
 | 7        | 7          | 1.7          | 07-Jan-2000 | zzzzzz7  |
 | 6        | 6          | 1.6          | 06-Jan-2000 | zzzzzz6  |
 | 5        | 5          | 1.5          | 05-Jan-2000 | zzzzzz5  |
 | 4        | 4          | 1.4          | 04-Jan-2000 | zzzzzz4  |
 | 3        | 3          | 1.3          | 03-Jan-2000 | zzzzzz3  |
 +----------+------------+--------------+-------------+----------+



mSQL > 
Query buffer
------------
select text_val from limit_test limit 10
[continue]
    -> 
Query OK.  10 row(s) modified or retrieved.

 +----------+
 | text_val |
 +----------+
 | zzzzzz1  |
 | zzzzzz2  |
 | zzzzzz3  |
 | zzzzzz4  |
 | zzzzzz5  |
 | zzzzzz6  |
 | zzzzzz7  |
 | zzzzzz8  |
 | zzzzzz9  |
 | zzzzzz10 |
 +----------+



mSQL > 
Query buffer
------------
select text_val from limit_test limit 100
[continue]
    -> 
Query OK.  29 row(s) modified or retrieved.

 +----------+
 | text_val |
 +----------+
 | zzzzzz1  |
 | zzzzzz2  |
 | zzzzzz3  |
 | zzzzzz4  |
 | zzzzzz5  |
 | zzzzzz6  |
 | zzzzzz7  |
 | zzzzzz8  |
 | zzzzzz9  |
 | zzzzzz10 |
 | zzzzzz11 |
 | zzzzzz12 |
 | zzzzzz13 |
 | zzzzzz14 |
 | zzzzzz15 |
 | zzzzzz16 |
 | zzzzzz17 |
 | zzzzzz18 |
 | zzzzzz19 |
 | zzzzzz20 |
 | zzzzzz21 |
 | zzzzzz22 |
 | zzzzzz23 |
 | zzzzzz24 |
 | zzzzzz25 |
 | zzzzzz26 |
 | zzzzzz27 |
 | zzzzzz28 |
 | zzzzzz29 |
 +----------+



mSQL > 
Bye!

